### PR TITLE
SQL: ignore UNSUPPORTED fields for JDBC and ODBC modes in 'SYS COLUMNS'

### LINKED ISSUE #39471: SQL: hide the geo_point columns for ODBC
The columns of type `geo_point` should also be hidden in `SYS COLUMNS` for the `ODBC` clients, just like `nested` and `object` (similar to https://github.com/elastic/elasticsearch/issues/37801).

### PR DESCRIPTION
**Pull request by astefan**
Up until now NESTED and OBJECT fields were ignored in SYS COLUMNS for ODBC mode only.

This PR changes this behavior and ES SQL will also ignore UNSUPPORTED fields in `SYS COLUMNS`.
Further more, it will ignore the same NESTED, OBJECT and UNSUPPORTED fields for JDBC mode, as well.

Fixes https://github.com/elastic/elasticsearch/issues/39471.

### CODE REVIEW

File: x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/plan/logical/command/sys/SysColumns.java [hunk 1]
```diff
@@ -185,6 +185,10 @@ private static Object odbcCompatible(Integer value, boolean isOdbcClient) {
         }
         return value;
     }
+    
+    private static boolean isOdbcClient(Mode mode) {
```
Comment (costin) [inline x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/plan/logical/command/sys/SysColumns.java]: No need to add a method for a simple check that is only used once.

**Review by costin — APPROVED** [review]
LGTM

File: x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/plan/logical/command/sys/SysColumns.java (hunk 1 above)
Comment (astefan) [reply x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/plan/logical/command/sys/SysColumns.java]: This method is used twice actually. [Here](https://github.com/elastic/elasticsearch/pull/39518/files#diff-8dd83d03e8d8b05d260314f0b99c40d0R100) as well.

**Comment by astefan** [comment]
@elasticmachine run elasticsearch-ci/2

File: x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/plan/logical/command/sys/SysColumns.java (hunk 1 above)
Comment (costin) [reply x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/plan/logical/command/sys/SysColumns.java]: My point still stands - it's a basic check.

**Review by matriv — APPROVED** [review]
LGTM!

**Pull request merged**